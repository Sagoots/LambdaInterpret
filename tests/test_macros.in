#define ID = \x.x
ID A
#define SELF_APPLY = \f.f f
SELF_APPLY ID
#define CONST_FIVE = \x.5
CONST_FIVE A

Y (\f x.x)