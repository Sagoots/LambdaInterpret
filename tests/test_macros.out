--- λ-Calculus REPL ---
Input from: C:\Users\solom\Documents\Wolfram\lambdaREPL\tests\test_macros.in
Output to: C:\Users\solom\Documents\Wolfram\lambdaREPL\tests\test_macros.out
Enter expressions or macro definitions (e.g., #define ID <expression>).
 Type 'exit' or 'quit' to end.
λ> Syntax error: Unexpected token MyLambdaInterpreter`Private`equal at atom parsing

λ> ID A

λ> Syntax error: Unexpected token MyLambdaInterpreter`Private`equal at atom parsing

λ> SELF_APPLY ID

λ> Syntax error: Unexpected token MyLambdaInterpreter`Private`equal at atom parsing

λ> CONST_FIVE A

λ> Y (λf x.x) →
(λf.(λx.f (x x)) (λx.f (x x))) (λf x.x) →
(λx.(λf x.x) (x x)) (λx.(λf x.x) (x x)) →
(λf x.x) ((λx.(λf x.x) (x x)) (λx.(λf x.x) (x x))) →
λx.x

λ> End of input or input canceled. Exiting REPL.
